catalog/core/catalog-core-metricsplugin/src/test/java/ddf/catalog/metrics/CatalogMetricsTest.java

```diff
@@ -33,146 +33,119 @@
-    // up the log to point of Travis CI build failing
-
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE, "TotalResults"));
-
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE, "Federated"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE, "Comparison"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE, "Spatial"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE, "Fuzzy"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE, "Temporal"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.QUERIES_SCOPE, "Function"));
-
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.EXCEPTIONS_SCOPE));
-    underTest.metrics.remove(
-        MetricRegistry.name(CatalogMetrics.EXCEPTIONS_SCOPE, "UnsupportedQuery"));
-    underTest.metrics.remove(
-        MetricRegistry.name(CatalogMetrics.EXCEPTIONS_SCOPE, "SourceUnavailable"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.EXCEPTIONS_SCOPE, "Federation"));
-
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.INGEST_SCOPE, "Created"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.INGEST_SCOPE, "Updated"));
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.INGEST_SCOPE, "Deleted"));
-
-    underTest.metrics.remove(MetricRegistry.name(CatalogMetrics.RESOURCE_SCOPE));
-
-    underTest.reporter.stop();
-  }
-
   @Test
-  public void catalogQueryMetric() throws Exception {
-    QueryRequest query = new QueryRequestImpl(new QueryImpl(idFilter));
-    underTest.process(query);
-
-    assertThat(underTest.queries.getCount(), is(1L));
-    assertThat(underTest.comparisonQueries.getCount(), is(1L));
+  public void testNullFilterAdapter() {
+    exception.expect(NullPointerException.class);
+    exception.expectMessage("filterAdapter");
+    new CatalogMetrics(null);
   }
 
   @Test
-  public void catalogFederatedQueryMetric() throws Exception {
-    QueryRequest query = new QueryRequestImpl(new QueryImpl(idFilter), true);
-    underTest.process(query);
-
-    query = new QueryRequestImpl(new QueryImpl(idFilter), Arrays.asList("fedSourceId"));
-    underTest.process(query);
+  public void testSourceComparisonQuery() throws Exception {
+    Iterable<Tag> tags = Tags.of("source", "testSite");
+    QueryRequest query =
+        new QueryRequestImpl(new QueryImpl(idFilter), Collections.singleton("testSite"));
+    catalogMetrics.process(query);
 
-    query =
-        new QueryRequestImpl(
-            new QueryImpl(idFilter), Arrays.asList("fedSource1Id", "fedSource2Id"));
-    underTest.process(query);
-
-    assertThat(underTest.federatedQueries.getCount(), is(3L));
+    assertThat(meterRegistry.counter("ddf.catalog.query.comparison", tags).count(), is(1.0));
```

Comment:
I tried to come up with a way to verify this without adding too much logic or coupling to unrelated implementation details.

For example, `assertThat(meterRegistry.getMeters().size(), is(1));` failed for most tests. The values were all over the place depending on unrelated metrics that were recorded while trying to test for one specific metric.

Based on that I think checking for the other metrics would be too brittle.

Reply:
Ah right because it lives in static space so all the other stuff is still getting reported. Gotcha. What if we just checked the specific component under test to ensure it was zero before we ran the plugin?

Reply:
I create a new registry in setup so the metrics are always cleared for each test.